File: swea/D3/11688.Calkin-Wilf tree/11688.py
T = int(input())
for tc in range(1, T+1):
    cmd = input()

    a, b = 1, 1
    for i in cmd:
        if i == 'L':
            b += a
        else:
            a += b
    
    print(f'#{tc} {a} {b}')

File: swea/D3/11688.Calkin-Wilf tree/11688.cpp
#include <iostream>
#include <string>

using namespace std;

int T;

int main(){
  cin >> T;
  for(int t = 1; t <= T; t++){
    string cmd;
    cin >> cmd;

    int a = 1, b = 1;
    for(int i = 0; i < cmd.length(); i++){
      if(cmd[i] == 'L') b += a;
      else a += b;
    }
    
    cout << '#' << t << ' ' << a << ' ' << b << endl;
  }
}